### PR TITLE
feat: add Fixes Inscription Protocol on Flow

### PR DESCRIPTION
**Pull request by btspoony**
##### Name (to be shown on DefiLlama): 

FIXeS World

##### Twitter Link:

https://twitter.com/fixesOnFlow

##### List of audit links if any:

N/A

##### Website Link:

https://fixes.world/

##### Logo (High resolution, will be shown with rounded borders):

FIXeS Logo-512x512

##### Current TVL:

129.56K $FLOW
 
##### Treasury Addresses (if the protocol has treasury)

https://www.flowdiver.io/account/0xd2abb5dbf5e08666

_This is the contract address, $FLOW in Treasury is locked inside contract account's resources_ 

##### Chain:

Flow

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)

N/A

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)

N/A

##### Short Description (to be shown on DefiLlama):

FIXeS is an executable inscription protocol on Flow blockchain. It allows users to inscribe anything on the blockchain and also provides the 𝔉rc20 fungible tokens.

##### Token address and ticker if any:

Official Ticker: $FLOWS, $FIXES

##### Category (full list at https://defillama.com/categories) *Please choose only one:

fixes-frc20: [Services](https://defillama.com/protocols/Services)
fixes-frc20-staking: [Staking Pool](https://defillama.com/protocols/Staking%20Pool)

##### forkedFrom (Does your project originate from another project):

Original

##### methodology (what is being counted as tvl, how is tvl being calculated):

TVL Methodology:

fixes-frc20 
- Treasury Pool: When minting 𝔉rc20 fungible tokens, $FLOW (minting cost) will be deposited into the Treasury Pool. If 𝔉rc20 is a burnable token, users can retrieve $FLOW from the Treasury Pool through the **burn** operation.
The minting cost is being counted as tvl.

fixes-frc20-staking
- Staking Pool: Users can stake their 𝔉rc20 tokens to receive benefits, such as fee sharing and 𝔉rc20 token vesting. Currently, only $FLOWS staking is open.
The staked $FLOWS is being counted as tvl.

Docs: <https://docs.fixes.world/>

##### Github org/user (Optional, if your code is open source, we can track activity):

<https://github.com/fixes-world/>

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20-staking exports TVL: 
        
 
 
    flow                      36.12 k
    
    total                    36.12 k

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20 exports TVL: 
        
 
 
    flow                      93.51 k
    
    total                    93.51 k

**Comment by g1nt0ki** [comment]
hi @btspoony thanks for the PR

Counting the treasury value as tvl as they can be burned to get back the FLOW makes sense, but I dont agree with liquid staking part, these are tokens minted by your protocol whose value is already counted as tvl, also, Liquid staking is meant for staking derivative like Lido ETH for ETH, i.e native/gas token of chain is locked/staked via some protocol, dont think that applies to your case.

I can merge this PR after you remove "projects/fixes-frc20-staking"?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
.

**Comment by btspoony** [comment]
> hi @btspoony thanks for the PR
> 
> Counting the treasury value as tvl as they can be burned to get back the FLOW makes sense, but I dont agree with liquid staking part, these are tokens minted by your protocol whose value is already counted as tvl, also, Liquid staking is meant for staking derivative like Lido ETH for ETH, i.e native/gas token of chain is locked/staked via some protocol, dont think that applies to your case.
> 
> I can merge this PR after you remove "projects/fixes-frc20-staking"?

hi @g1nt0ki 

Staking Part is indeed the Token minted by my protocol.

But in the Stake operation, they are locked again (cannot circulate on the Market, but users can obtain a SemiNFT as proof).
Because of the existence of the Market, these Tokens have a token value(Floor Price) higher than the Treasury Pool value.

---

Can I count the `(Market Token Value - Treasure Pool Benchmark Value) * Staked Amount` as TVL?

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20-staking exports TVL: 
        
 
 
    flow                      50.60 k
    
    total                    50.60 k

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20 exports TVL: 
        
 
 
    flow                      102.78 k
    
    total                    102.78 k

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20-staking exports TVL: 
        
 
 
    flow                      50.60 k
    
    total                    50.60 k

**Comment by llamatester** [comment]
The adapter at projects/fixes-frc20 exports TVL: 
        
 
 
    flow                      102.78 k
    
    total                    102.78 k

**Comment by btspoony** [comment]
> hi @btspoony thanks for the PR
> 
> Counting the treasury value as tvl as they can be burned to get back the FLOW makes sense, but I dont agree with liquid staking part, these are tokens minted by your protocol whose value is already counted as tvl, also, Liquid staking is meant for staking derivative like Lido ETH for ETH, i.e native/gas token of chain is locked/staked via some protocol, dont think that applies to your case.
> 
> I can merge this PR after you remove "projects/fixes-frc20-staking"?



> hi @btspoony thanks for the PR
> 
> Counting the treasury value as tvl as they can be burned to get back the FLOW makes sense, but I dont agree with liquid staking part, these are tokens minted by your protocol whose value is already counted as tvl, also, Liquid staking is meant for staking derivative like Lido ETH for ETH, i.e native/gas token of chain is locked/staked via some protocol, dont think that applies to your case.
> 
> I can merge this PR after you remove "projects/fixes-frc20-staking"?

Changed to the new methodology I mentioned, please take a look and see if this can be considered as TVL.
It subtracted the portion in Treasury Pool that is already calculated as TVL (which is Benchmark Value).

File: projects/fixes-frc20-staking/index.js [hunk 1]
```diff
@@ -52,7 +53,7 @@ fun main(): UFix64 {
         var details = stakingPool!.getDetails()
         let validStaked = details.totalStaked - details.totalUnstakingLocked
 
-        totalTVL = totalTVL + (validStaked * floorPrice)
+        totalTVL = totalTVL + (validStaked * (floorPrice - benchmarkPrice))
```
Comment (btspoony) [inline projects/fixes-frc20-staking/index.js:56]: New Staking TVL Methodolgy:
TVL: validStaked * (floorPrice - benchmarkPrice)

**Pull request merged**